### ISSUE TITLE
Add temporary / custom country codes

### ISSUE THREAD

**Issue by eliasmistler** [pycountry/pycountry#109]
Hi there,

Really appreciate the library and have been using it happily for a while, until today I stumbled on a problem: I'm working on data for Kosovo and need to store said data against an ISO `alpha_2` code.

# Background

Kosovo is recognised as an independent country by 97 UN member states, which is a considerable amount but not a majority. As a consequence, there is no "official" iso-3166-1 code, however the keys `XK`, `XXK` and `926` are commonly used as temporary, or user-assigned codes.

As per the [Wikipedia page](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#XK), section on "User-assigned code elements":

> The code `XK` is being used by the [European Commission](https://en.wikipedia.org/wiki/European_Commission),[[26]](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#cite_note-26) the [IMF](https://en.wikipedia.org/wiki/International_Monetary_Fund), and [SWIFT](https://en.wikipedia.org/wiki/Society_for_Worldwide_Interbank_Financial_Telecommunication),[[27]](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#cite_note-27) the [CLDR](https://en.wikipedia.org/wiki/Common_Locale_Data_Repository), and other organizations as a temporary country code for [Kosovo](https://en.wikipedia.org/wiki/Kosovo)

# Issue / Feature Request

To my understanding, `pycountry` currently does not come with any user-assigned or temporary codes, nor does it have a public API to add to any of the collections. Please correct me if I've missed something.

I see two possible solutions to the issue above:
- Create an additional collection (like `historic_countries`) of temporary and commonly used user-assigned codes
- Allow users to add to a collection via a public API like `pycountry.countries.add(alpha_2="XK", alpha_3="XXK", name="Kosovo", numeric="926")

# Temporary Solution
I have solved the issue as follows in my code base, deriving my code from `pycountry.db.Database._load`

```python
import pycountry

def add_to_pycountry(database: pycountry.db.Database, **entry) -> None:
    database._load()

    # create the object with the correct dynamic type
    cls = type(database.objects[0])
    obj = cls(**entry)

    # append object
    database.objects.append(obj)

    # update indices
    for key, value in entry.items():
        value = value.lower()
        if key in database.no_index:
            continue
        index = database.indices.setdefault(key, {})
        index[value] = obj


add_to_pycountry(pycountry.countries, alpha_2="XK", alpha_3="XXK", name="Kosovo", numeric="926")
```

I would really appreciate a solution in the library itself, rather than having to work around it. I'd be happy to submit a PR if useful.

Thanks,
Elias

**Comment by nschimmoller** [comment]
@eliasmistler can't hurt to submit a PR :)

**Comment by ctheune** [comment]
Yeah, that's definitely a good idea. I always considerd pycountry to be flexible enough to massage the data as you need, but this is like a good piece for the roadmap to make official APIs for adding,deleting and updating records on the fly.

**Comment by eliasmistler** [comment]
Sorry it's taken so long, but I've finally submitted a [PR](https://github.com/flyingcircusio/pycountry/pull/145) addressing this. I didn't just want to rush this in, but first understand the rationale and style of the library, adjust my code to it, and make it useful in a wider sense. I believe I have achieved that in the PR -- looking forward to your review!

**Comment by eliasmistler** [comment]
@ctheune any feedback on this PR? It's been sitting open for a while now, so just wanted to check in if it needs any adjustment before you'd consider merging it.

**Comment by lfg6000** [comment]
@eliasmistler thanks for the add_to_pycountry() method.  kosovo appeared and was causing a crash in the app.  applied your fix to get the app running again. thanks.

**Comment by nschimmoller** [comment]
@eliasmistler just merged this as part of #175 thanks so much for your contributions! Will talk to other maintainers about a PyPI release

**Comment by Chuukwudi** [comment]
This only updates the indices. I wish it creates a new one. For instance, I want both"Brazil" and "Brasil" to be correctly identified as `Country(alpha_2='BR', alpha_3='BRA', flag='🇧🇷', name='Brazil', numeric='076', official_name='Federative Republic of Brazil')` unfortunately, adding a new entry overwrites the previous one.
